src/Chainweb/Sync/ForkInfo.hs

```diff
@@ -164,28 +165,40 @@ resolveForkInfoForProviderState logg bhdb candidateHdrs provider hints finfo ppK
         logg Info $ "resolveForkInfo: payload provider state: " <> brief ppKnownState
             <> "; target state: " <> brief (_forkInfoTargetState finfo)
 
+        let lookupBhdb rbh = lookupRanked bhdb (int $ _rankedHeight rbh) (_ranked rbh)
+
         hdr :: BlockHeader <- runMaybeT
-            (MaybeT (tableLookup candidateHdrs (_ranked trgHash))
-                <|> MaybeT (lookupRanked bhdb (int $ _rankedHeight trgHash) (_ranked trgHash)))
+            (MaybeT (tableLookup candidateHdrs (_ranked trgHash)) <|> MaybeT (lookupBhdb trgHash))
                 >>= \case
                     Nothing -> do
-                        throwM $ InternalInvariantViolation $ "validatePayload: target block is missing: " <> brief trgHash
+                        throwM $ InternalInvariantViolation $
+                            "validatePayload: target block is missing: " <> brief trgHash
                     Just hdr -> return hdr
 
+        -- Finding a fork from the latest block can fail if we don't know the
+        -- state of the payload provider.
+        -- This can happen if:
+        --   1. the payload provider db comes from an external source.
+        --   2. the payload provider has moved to a fork that catchup has not
+        --   finished validating.
+        -- In this case we can only guess where a common fork point might be --
+        -- in the worst case this is genesis. The more common case is 2. in
+        -- which case the "safe" block is probably known.
```

Comment:
```suggestion
        -- In this case we can only guess where a common fork point might be --
        -- in the worst case this is the final block. The more common case is 2. in
        -- which case the "safe" block is probably known. Only in exceptional case
        -- this would be Genesis.
```